### PR TITLE
swapped #include <fms_platform.h> with use :: platform_mod, and updat…

### PR DESCRIPTION
**Pull request by wrongkindofdoctor**
**Description**
* updated the rest of the makefile.am files in the FMS repos to include platform.F90
* replaced platform.h with platform.F90 types in all FMS files. It was necessary to do this all at once because everything depends on mpp.F90.
* doxygenized and moved comments to spaces adjacent to routines in some MPP modules

**How Has This Been Tested?**
tested on Gaea with intel19 and gcc 8.3, Skylake with intel19 and gcc 9.3
**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [O] `make distcheck` passes (3 fms2-io tests fail due to segfaults when performing checksums on i8_kind data; the same errors occur in the main branch).

### CODE REVIEW

**Review by GFDL-Eric — APPROVED** [review]
Intense!

**Comment by ngs333** [comment]
This changes over a hundred files! :-(

File: mpp/include/mpp_define_nest_domains.inc [hunk 1]
```diff
@@ -21,20 +21,89 @@
...
+!!    call mpp_get_F2C_index(nest_domain, is_c, ie_c, js_c, je_c, is_f, ie_f, js_f, je_f, nest_level=level)\n
+!!    allocate(buffer (is_f:ie_f, js_f:je_f,nz))\n
+!!    call mpp_update_nest_coarse(x, nest_domain, buffer)
+!
+!> @example all mpp_define_nest_domains (nest_domain, domain, num_nest, nest_level, tile_fine, tile_coarse, &\n
+!!                               istart_coarse, icount_coarse, jstart_coarse, jcount_coarse, &\n
+!!                                    npes_nest_tile, x_refine, y_refine, extra_halo, name)
+!
+!> @note Currently the contact will be limited to overlap contact.
+
```
Comment (ngs333) [inline mpp/include/mpp_define_nest_domains.inc:82]: This was a lot of work!

File: mpp/mpp_domains.F90 [hunk 2]
```diff
@@ -17,105 +17,51 @@
...
+!! the second index, treated as a special case of 2D decomposition, is
+!! also possible. We define domain as the grid associated with a <I>task</I>.
+!! We define the compute domain as the set of gridpoints that are
+!! computed by a task, and the data domain as the set of points
+!! that are required by the task for the calculation. There can in
+!! general be more than 1 task per PE, though often
+!! the number of domains is the same as the processor count. We define
+!! the global domain as the global computational domain of the
+!! entire model (i.e, the same as the computational domain if run on a
+!! single processor). 2D domains are defined using a derived type domain2,
```
Comment (ngs333) [inline mpp/mpp_domains.F90:61]: Don't you mean "domain2D"  and not "domain2"

File: mpp/mpp_domains.F90 [hunk 3]
```diff
@@ -17,105 +17,51 @@
...
+!! update API for rectilinear grids, built on top of the mpp_mod API for message passing.
+!! Features of mpp_domains_mod include:\n
+!!\n
+!! Simple, minimal API, with free access to underlying API for more complicated stuff.\n
+!!\n
+!! Design toward typical use in climate/weather CFD codes.
+!!
+!> @par[Domains]
+!!
+!! It is assumed that domain decomposition will mainly be in 2
```
Comment (ngs333) [inline mpp/mpp_domains.F90:48]: Hmmm!

File: mpp/mpp_domains.F90 [hunk 4]
```diff
@@ -17,105 +17,51 @@
...
+!! also possible. We define domain as the grid associated with a <I>task</I>.
+!! We define the compute domain as the set of gridpoints that are
+!! computed by a task, and the data domain as the set of points
+!! that are required by the task for the calculation. There can in
+!! general be more than 1 task per PE, though often
+!! the number of domains is the same as the processor count. We define
+!! the global domain as the global computational domain of the
+!! entire model (i.e, the same as the computational domain if run on a
+!! single processor). 2D domains are defined using a derived type domain2,
+!! constructed as follows (see comments in code for more details).
```
Comment (ngs333) [inline mpp/mpp_domains.F90:62]: "constructed as follows". No comments follow. Did you accidentally delete comments?

File: mpp/mpp_efp.F90 [hunk 5]
```diff
@@ -236,24 +235,22 @@ function mpp_reproducing_sum_r4_2d(array, isr, ier, jsr, jer, EFP_sum, reproduci
 
 end function mpp_reproducing_sum_r4_2d
 
-
+!> This functioon uses a conversion to an integer representation
```
Comment (ngs333) [inline mpp/mpp_efp.F90:238]: function

Comment (ngs333) [reply mpp/mpp_efp.F90:238]: Maybe its Slovakian?

**Review by ngs333 — APPROVED** [review]

File: mpp/include/mpp_define_nest_domains.inc (hunk 1 above)
Comment (wrongkindofdoctor) [reply mpp/include/mpp_define_nest_domains.inc:82]: It was the equivalent of scrubbing grout with a toothbrush. I will resolve the typos and look into the potentially missing example (copy-paste FAIL).

**Comment by ngs333** [comment]
You need an electric toothbrush!

File: fms2_io/Makefile.am [hunk 6]
```diff
@@ -38,8 +38,9 @@ include/register_variable_attribute.inc					\
 include/unstructured_domain_write.inc
 
 # Include .h and .mod files.
-AM_CPPFLAGS = -I${top_builddir}/mpp
-AM_CPPFLAGS += -I${top_srcdir}/fms2_io/include
+AM_CPPFLAGS = -I${top_builddir}/mpp \
```
Comment (laurenchilutti) [inline fms2_io/Makefile.am:41]: Is there a reason you change the style of this here and in mpp/Makefile.am? In all others you just append (AM_CPPFLAGS += ...)?

**Review by mlee03 — APPROVED** [review]

File: mpp/mpp_domains.F90 [hunk 7]
```diff
@@ -365,60 +313,63 @@ module mpp_domains_mod
...
+     type(overlapSpec),  pointer :: update_T      => NULL() !< send and recv information for halo update of T-cell.
+     type(overlapSpec),  pointer :: update_E      => NULL() !< send and recv information for halo update of E-cell.
+     type(overlapSpec),  pointer :: update_C      => NULL() !< send and recv information for halo update of C-cell.
+     type(overlapSpec),  pointer :: update_N      => NULL() !< send and recv information for halo update of N-cell.
+     type(domain2d),     pointer :: io_domain     => NULL() !< domain for IO, will be set through calling mpp_set_io_domain ( this will be changed).
   end type domain2D
 
-  !--- the following type is used to reprsent the contact between tiles.
-  !--- this type will only be used in mpp_domains_define.inc
+  !> Type used to reprsent the contact between tiles.
```
Comment (laurenchilutti) [inline mpp/mpp_domains.F90:361]: Here's an old typo -  "represent"

File: fms2_io/Makefile.am (hunk 6 above)
Comment (laurenchilutti) [reply fms2_io/Makefile.am:41]: Nothing necessarily needs to be updated with this comment - I'm just curious which method is preferred

**Review by laurenchilutti — APPROVED** [review]

File: fms2_io/Makefile.am (hunk 6 above)
Comment (wrongkindofdoctor) [reply fms2_io/Makefile.am:41]: Just personal preference; I like the streamlined appearance of newlines, and it's slightly easier to append paths as needed. Use whatever floats your boat.

File: mpp/mpp_domains.F90 (hunk 4 above)
Comment (wrongkindofdoctor) [reply mpp/mpp_domains.F90:62]: I re-inserted the example datatype structs.

File: mpp/mpp_domains.F90 (hunk 2 above)
Comment (wrongkindofdoctor) [reply mpp/mpp_domains.F90:61]: Fixed

**Pull request merged**